test/tinyRocket_nangate45.metrics

```diff
@@ -3,67 +3,69 @@
-	"route__wirelength__estimated": 479089,
-	"RSZ::repair_design_buffer_count": "515",
-	"RSZ::max_slew_slack": "40.53570401918181",
+	"design__io__hpwl": 37337478,
+	"design__instance__displacement__total": 25734.7,
+	"design__instance__displacement__mean": 1.0775,
+	"design__instance__displacement__max": 9.722,
+	"route__wirelength__estimated": 480419,
+	"RSZ::repair_design_buffer_count": "524",
+	"RSZ::max_slew_slack": "32.36975340786154",
 	"RSZ::max_fanout_slack": "100.0",
-	"RSZ::max_capacitance_slack": "35.658236045488195",
-	"design__instance__displacement__total": 437.308,
+	"RSZ::max_capacitance_slack": "34.74309796237679",
+	"design__instance__displacement__total": 435.337,
 	"design__instance__displacement__mean": 0.018,
-	"design__instance__displacement__max": 15.754,
-	"route__wirelength__estimated": 492762,
+	"design__instance__displacement__max": 4.015,
+	"route__wirelength__estimated": 494083,
 	"design__instance__count__setup_buffer": 0,
 	"design__instance__count__hold_buffer": 0,
-	"RSZ::worst_slack_min": "0.06525749569871833",
-	"RSZ::worst_slack_max": "0.2572222507796264",
+	"RSZ::worst_slack_min": "0.07174036567860885",
+	"RSZ::worst_slack_max": "0.1901969746643768",
 	"RSZ::tns_max": "0.0",
 	"RSZ::hold_buffer_count": "0",
 	"design__instance__displacement__total": 0,
 	"design__instance__displacement__mean": 0,
 	"design__instance__displacement__max": 0,
-	"route__wirelength__estimated": 492762,
+	"route__wirelength__estimated": 494083,
 	"DPL::utilization": "25.4",
-	"DPL::design_area": "58353",
-	"route__net": 27908,
+	"DPL::design_area": "58357",
+	"route__net": 27918,
 	"route__net__special": 2,
 	"antenna__violating__nets": 0,
 	"antenna__violating__pins": 0,
 	"GRT::ANT::errors": "0",
-	"design__violations": 0,
-	"route__net": 27908,
+	"route__net": 27918,
 	"route__net__special": 2,
-	"route__drc_errors__iter:1": 7055,
-	"route__wirelength__iter:1": 598035,
-	"route__drc_errors__iter:2": 264,
-	"route__wirelength__iter:2": 595628,
-	"route__drc_errors__iter:3": 71,
-	"route__wirelength__iter:3": 595453,
-	"route__drc_errors__iter:4": 4,
-	"route__wirelength__iter:4": 595447,
+	"route__drc_errors__iter:1": 2875,
+	"route__wirelength__iter:1": 600462,
+	"route__drc_errors__iter:2": 247,
+	"route__wirelength__iter:2": 598431,
+	"route__drc_errors__iter:3": 117,
+	"route__wirelength__iter:3": 598372,
+	"route__drc_errors__iter:4": 3,
+	"route__wirelength__iter:4": 598382,
 	"route__drc_errors__iter:5": 0,
-	"route__wirelength__iter:5": 595447,
+	"route__wirelength__iter:5": 598385,
 	"route__drc_errors": 0,
-	"route__wirelength": 595447,
-	"route__vias": 181692,
-	"route__vias__singlecut": 181692,
+	"route__wirelength": 598385,
+	"route__vias": 181888,
+	"route__vias__singlecut": 181888,
 	"route__vias__multicut": 0,
 	"DRT::drv": "0",
 	"antenna__violating__nets": 0,
 	"antenna__violating__pins": 0,
+	"antenna__violating__nets": 0,
+	"antenna__violating__pins": 0,
 	"DRT::ANT::errors": "0",
```

Comment:
Why are there duplication metrics?

Reply:
I didn't notice that. I thought metrics with the same name would be overwritten when writing it multiple times, but I guess it just creates multiple entries now that we call check_antennas in the loop.

Reply:
No it just uses a list of metrics.  You could append the iteration number to make them unique.

Reply:
Can this be done in the flow.tcl script? I don't have access to the iteration number in the C++ implementation.

Reply:
Why do we have two iterations if the violations are solved in the first one?

Reply:
In this case, I believe the problem is because we're calling check_antennas in three different places:
1. After global_route
2. During the iterations to repair_antennas post-DRT
3. After the iterations, for the final report.

In the first case, we call it to generate the report for post-GRT violations.

The third case might be redundant in cases where all the violations are solved in less than 5 iterations, but it is necessary for the cases where some iterations remains.

Reply:
Perhaps can be used utl::set_metrics_stage to solve this?

Reply:
Thanks! I will give it a try.